src/shared/scrapers/US/CA/san-francisco-county.js

```diff
@@ -9,26 +9,99 @@ const scraper = {
-  url: 'https://www.sfdph.org/dph/alerts/coronavirus.asp',
-  type: 'paragraph',
-  async scraper() {
-    let deaths;
-    let cases;
-    const $ = await fetch.page(this, this.url, 'default');
-    const $h2 = $('h2:contains("Cases in San Francisco")');
+  maintainers: [maintainers['1ec5']],
+  type: 'json',
+  timeseries: true,
+  headless: false,
+  sources: [
     {
-      const $p = $h2.nextAll('*:contains("Cases:")');
-      cases = parse.number($p.text());
-    }
+      name: 'San Francisco Department of Public Health',
+      url: 'https://data.sfgov.org/COVID-19/COVID-19-Cases-Summarized-by-Date-Transmission-and/tvq9-ec9w',
+      description: 'COVID-19 Cases Summarized by Date, Transmission and Case Disposition'
+    },
+    {
+      name: 'San Francisco Department of Public Health',
+      url: 'https://data.sfgov.org/COVID-19/COVID-19-Hospitalizations/nxjg-bhem',
+      description: 'COVID-19 Hospitalizations'
+    },
     {
-      const $p = $h2.nextAll('*:contains("Deaths:")');
-      deaths = parse.number($p.text());
+      name: 'San Francisco Department of Public Health',
+      url: 'https://data.sfgov.org/COVID-19/COVID-19-Tests/nfpa-mg4g',
+      description: 'COVID-19 Tests'
+    }
+  ],
+  url: 'https://data.sfgov.org/stories/s/dak2-gvuj',
+  _urls: {
+    cases: 'https://data.sfgov.org/resource/tvq9-ec9w.json',
+    patients: 'https://data.sfgov.org/resource/nxjg-bhem.json',
+    tests: 'https://data.sfgov.org/resource/nfpa-mg4g.json'
+  },
+  scraper: {
+    '0': async function() {
+      const cases = await fetch.json(this, this._urls.cases, 'cases', false);
+      const patients = await fetch.json(this, this._urls.patients, 'patients', false);
+      const tests = await fetch.json(this, this._urls.tests, 'tests', false);
+      const timeSeries = {};
+      cases.reduceRight(
+        (acc, cur) => {
+          const date = datetime.getYYYYMMDD(cur.date);
+          const field = cur.case_disposition === 'Death' ? 'deaths' : 'cases';
+          acc[field] += +cur.case_count;
+          if (!(date in timeSeries)) {
+            timeSeries[date] = {};
+          }
+          timeSeries[date][field] = acc[field];
+          return acc;
+        },
+        {
+          cases: 0,
+          deaths: 0
+        }
+      );
+      patients.forEach(elt => {
+        const date = datetime.getYYYYMMDD(elt.reportdate);
+        if (!(date in timeSeries)) {
+          timeSeries[date] = {};
+        }
+        if (elt.patientcount) {
+          if (!('hospitalized' in timeSeries[date])) {
+            timeSeries[date].hospitalized = 0;
+          }
+          timeSeries[date].hospitalized += +elt.patientcount;
+        }
+      });
+      tests.reduceRight((acc, cur) => {
+        const date = datetime.getYYYYMMDD(cur.result_date);
+        acc += +cur.tests;
+        if (!(date in timeSeries)) {
+          timeSeries[date] = {};
+        }
+        timeSeries[date].tested = acc;
+        return acc;
+      }, 0);
+
```

Comment:
The undocumented standard :-) for scraping in this project appears to be checking date bounds at the beginning, and filtering data before scraping.  At least, that's what it should be.  Your method works too, so it's good and no need to change it, just a note for the future!

Reply:
The data source only provides the number of new cases on each day, whereas this project seems to expect a cumulative number on each day. Calculating the cumulative number unfortunately requires reducing from the end of the array, but we could have the `reduceRight` call stop at the requested date if performance is a concern.